distributed/protocol/serialize.py

```diff
@@ -90,6 +90,20 @@ def register_serialization_family(name, dumps, loads):
 register_serialization_family("error", None, serialization_error_loads)
 
 
+def check_dask_serializable(x):
+    if type(x) in (list, set, tuple) and len(x):
+        return check_dask_serializable(next(iter(x)))
+    elif type(x) is dict and len(x):
+        return check_dask_serializable(next(iter(x.items()))[1])
+    else:
+        try:
+            dask_serialize.dispatch(type(x))
```

Comment:
Just a note, this requires that objects be registered with `dask_serialize` (as we don't check `cuda_serialize` for example). However we do register all CUDA objects with `dask_serialize` now and use this in other contexts (like [Dask-CUDA's spilling]( https://github.com/rapidsai/dask-cuda/blob/b5ae4fc872b057f9d71595fccaa6dfd61654e301/dask_cuda/device_host_file.py#L71 )). So this seems like a good enough check.

Reply:
Thanks for the note @jakirkham - The fact that you have cleaned up serialization and registered everything as `dask_serialize` definitely made my life easier here - Thanks! :)